src/AutoRest.CSharp.V3/CodeGen/ClientWriter.cs

```diff
@@ -176,53 +175,28 @@ private void WriteConstant(CodeWriter writer, ClientConstant constant)
             if (constant.Value == null)
             {
                 // Cast helps the overload resolution
-                writer.Append("(")
-                    .AppendType(_typeFactory.CreateType(constant.Type))
-                    .Append(")")
-                    .Literal(null);
+                writer.Append($"({_typeFactory.CreateType(constant.Type)}){null:L}");
                 return;
             }
 
             switch (constant.Type)
             {
                 case FrameworkTypeReference frameworkType when frameworkType.Type == typeof(DateTimeOffset):
-                    var dateTimeValue = (DateTimeOffset)constant.Value;
-                    dateTimeValue = dateTimeValue.ToUniversalTime();
-
-                    writer.Append("new ");
-                    writer.AppendType(typeof(DateTimeOffset));
-                    writer.Append("(");
-                    writer.Literal(dateTimeValue.Year);
-                    writer.Comma();
-                    writer.Literal(dateTimeValue.Month);
-                    writer.Comma();
-                    writer.Literal(dateTimeValue.Day);
-                    writer.Comma();
-                    writer.Literal(dateTimeValue.Hour);
-                    writer.Comma();
-                    writer.Literal(dateTimeValue.Minute);
-                    writer.Comma();
-                    writer.Literal(dateTimeValue.Second);
-                    writer.Comma();
-                    writer.Literal(dateTimeValue.Millisecond);
-                    writer.Comma();
-                    writer.AppendType(typeof(TimeSpan));
-                    writer.Append(".");
-                    writer.Append(nameof(TimeSpan.Zero));
-                    writer.Append(")");
+                    var d = (DateTimeOffset)constant.Value;
+                    d = d.ToUniversalTime();
+                    writer.Append($"new {typeof(DateTimeOffset)}({d.Year:L}, {d.Month:L}, {d.Day:L} ,{d.Hour:L}, {d.Minute:L}, {d.Second:L}, {d.Millisecond:L}, {typeof(TimeSpan)}.{nameof(TimeSpan.Zero)}");
```

Comment:
":L" format specifier renders parameter as literal.

passing typeof(bla) would also do the right thing